src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -1768,29 +1768,15 @@ unsupported {} from `{}` with element `{}` of size `{}` to `{}`"#,
 fn int_type_width_signed(ty: Ty, cx: &CodegenCx) -> Option<(u64, bool)> {
     match ty.sty {
         ty::TyInt(t) => Some((match t {
-            ast::IntTy::Isize => {
-                match &cx.tcx.sess.target.target.target_pointer_width[..] {
-                    "16" => 16,
-                    "32" => 32,
-                    "64" => 64,
-                    tws => bug!("Unsupported target word size for isize: {}", tws),
-                }
-            },
+            ast::IntTy::Isize => cx.tcx.sess.target.isize_ty.bit_width().unwrap() as u64,
```

Comment:
this isn't a pure refactoring ... might a mischevious target attempt to choose a bitwidth other than `{16,32,64}` ?

Having said that, I don't particularly mind either way

Reply:
(and furthermore, this new code is arguably much cleaner and more directly expressing intent.)

Reply:
The isize_ty is created via an equivalent match anyway, so I don't think it matters much.